logging/redaction/log_redaction.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import json
+import logging
+from typing import List
+
+from apache_beam import CombineFn, CombineGlobally, DoFn, io, ParDo, Pipeline, WindowInto
+from apache_beam.error import PipelineError
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.transforms.window import FixedWindows
+
+from google.cloud import logging_v2
+
+
+# TODO: Place inspection and de-identification configurations
+
+class PayloadAsJson(DoFn):
```

Comment:
These classes between `log_redaction` and `log_redaction_final` are identical - will they be different in the future?

Reply:
Not sure I understand the question. Files `log_redaction` and `log_redaction_final` serve as "starting" and "ending" of the tutorial. The tutorial instructs to use `log_redaction` and to apply step-by-step changes to achieve the end result. `log_redaction_final` is there in the case a reader wants to see (or use) the end result or for validation.
It makes all code present in `log_redaction` to be present in `log_redaction_final`.

Reply:
Got it. Thanks!